### ISSUE TITLE
feat: Improve Android background handling of data notifications

### ISSUE THREAD

**Issue by phal0r** [capawesome-team/capacitor-firebase#292]
**Is your feature request related to a problem? Please describe:**
Firebase already provides a service, that is extended by this plugin, for receiving data notifcations in the background. As services have a different lifecycle than activities, we can even receive data notifications, when the app is closed (not only in the background).

Currently the notification is stored as a static property of the plugin (https://github.com/capawesome-team/capacitor-firebase/blob/main/packages/messaging/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/messaging/FirebaseMessagingPlugin.java#L75)

**Describe the solution you'd like:**
I would like to be able to execute custom code in this case (in our case, we want to display a local notification). There are 2 possibilities to tackle this:

Solution A:
- define an interface, that can be implemented by a java class and allow specifying this class in the configuration (or define a reasonable default name, that must be used
- this class is checked for and called from the service in the case, the app is killed
- as developers we are now able to do whatever we want in processing the notification

Solution B:
- basically the same idea, but using JavaScriptSandbox and JavascriptIsolates to allow using JS in the service and not Java
- this is a similar approach as react native with its headlessjs system

**Describe alternatives you've considered:**
No alternatives as we can only execute code in the service, which is currently not possible without modifying the code of this plugin.

**Additional context:**

**Comment by Excel1** [comment]
This is also useful for translations, no? Cause in my case i receive the pushMessage (only data) and need to translate it cause the language is a local app setting. The received data will be transformed in a local push notification.

@phal0r can you present your workaround?

**Comment by Excel1** [comment]
For me everything works till the app got killed

```
    await FirebaseMessaging.addListener(
      'notificationReceived',
      async (event) => {
        console.log('Notification received: ', event);
        await LocalNotifications.schedule({
          notifications: [
            {
              title: event.notification.title,
              body: "Hello",
              id: 1,
              schedule: { at: new Date(Date.now()+ 2000) },
              smallIcon: 'ic_notification',
              actionTypeId: '',
              extra: null,
              iconColor: '#488AFF',
            },
          ],
        });
      },
    );
```

**Comment by phal0r** [comment]
@Excel1 yes, if the app is killed, you cannot execute any javascript anymore. Both iOS and Android have mechanisms to execute code in the background. So another approach is needed to be able to execute code in the background.

The capacitor team has this plugin: https://capacitorjs.com/docs/apis/background-runner#javascript-api
It has an API to schedule local notifications, which may be enough for your case (depending on where you store your translations).

However, for our use cases, the plugin is too limited. We are working (or let's say investigating) on a enhanced version, which allows generic capacitor plugins to work, here https://github.com/atroo/capacitor-background-runner-ng. This is a challenging task and finished yet, but imho worth the effort.

**Comment by Excel1** [comment]
@phal0r Thank you very much for your quick and detailed reply! 

As it is unfortunately only possible via capacitor in a roundabout way, I only send data notifications. My natively implemented Firebase messaging service displays them accordingly, even if the app has been killed. Accordingly, I have completely dispensed with Capacitor push notifications so that they all look the same.
